src/test/java/com/commercetools/sync/categories/utils/CategoryReferenceReplacementUtilsTest.java

```diff
@@ -98,14 +121,21 @@
         for (CategoryDraft referenceReplacedDraft : referenceReplacedDrafts) {
             assertThat(referenceReplacedDraft.getParent().getId()).isEqualTo(parentId);
             assertThat(referenceReplacedDraft.getCustom().getType().getId()).isEqualTo(customTypeId);
+
+            final List<AssetDraft> referenceReplacedDraftAssets = referenceReplacedDraft.getAssets();
+            assertThat(referenceReplacedDraftAssets).hasSize(1);
+            assertThat(referenceReplacedDraftAssets.get(0).getCustom()).isNotNull();
+            assertThat(referenceReplacedDraftAssets.get(0).getCustom().getType().getId())
+                .isEqualTo(asset.getCustom().getType().getId());
         }
     }
 
     @Test
     public void buildCategoryQuery_Always_ShouldReturnQueryWithAllNeededReferencesExpanded() {
         final CategoryQuery categoryQuery = CategoryReferenceReplacementUtils.buildCategoryQuery();
-        assertThat(categoryQuery.expansionPaths()).hasSize(2);
-        assertThat(categoryQuery.expansionPaths())
-                       .containsExactly(ExpansionPath.of("custom.type"), ExpansionPath.of("parent"));
+        assertThat(categoryQuery.expansionPaths()).containsExactly(
+            ExpansionPath.of("custom.type"),
+            ExpansionPath.of("assets[*].custom.type"),
```

Comment:
it seems it doesn't have sense to make unit tests for the expansion path, because it doesn't protect from typo mistakes and mistakes of incorrect expansion path describing/understanding.Actually, it should be an integration test where you fetch real data and verify the values on expanded objects

Reply:
well, the strings that are there in the assertions in the unit tests are the correct ones, so if someone changes the string in the actual method implementation, the test should fail. Of course, if someone changes in the unit test as well, then problem. But for any test it's like that, if you make a mistake in the assertion. 

A more typo proof one would be the integration test, correct. And this is coming in a separate PR as mentioned 😉